### DISCUSSION TITLE
Can I also set the temperature limit?

### DISCUSSION THREAD

**Discussion by YouHyin** [pybamm-team/PyBaMM#2320] [Q&A]
I am currently charging the cccv using pybamm.
At this time, the temperature and voltage limits should be put in this run. 
For voltage, it was put at an "until 4.1 V".
I want to set the temperature limit so that the temperature of the cell does not exceed 40 degrees.("Charge at 50 A until 4.1 V")
![온도한계](https://user-images.githubusercontent.com/113442395/192433717-c1b7bb7d-7b7e-4259-b1cc-3e25da059b12.png)
I'm really curious about what to do at this time.

I'd really appreciate it if you could help me.
Thank you!

**Comment by valentinsulzer** [comment]
You can do it like this (*not* using experiment).
```python3
import pybamm

pybamm.set_logging_level("INFO")

# load models
model = pybamm.lithium_ion.SPM({"thermal": "lumped"})
T = model.variables["Volume-averaged cell temperature [K]"]
model.events.append(pybamm.Event("Temperature cut-off", (273.15 + 30) - T))

param = pybamm.ParameterValues("Chen2020")
param.search("heat transfer coefficient")
param.update({"Total heat transfer coefficient [W.m-2.K-1]": 1})

# create and run simulations
sim = pybamm.Simulation(model, parameter_values=param, C_rate=-5)
sim.solve([0, 3600], initial_soc=0)

# plot
pybamm.dynamic_plot(sim, ["Volume-averaged cell temperature [K]"])
```
It's not possible when using experiment

**Reply by YouHyin** [reply]
Oh, thank you very much. It helped me a lot. I have one more question. I wanted to observe the temperature change over time with a plot with the code I posted in the picture. In theory, the temperature should change over time, but the plot shows that the temperature is constant over time. Is this right? What should I do if I want to graphically observe temperature changes over time?
![Figure_1](https://user-images.githubusercontent.com/113442395/192793448-9082e142-f765-4101-a75e-17ce3574a411.png)

**Reply by valentinsulzer** [reply]
You have to pass the options (e.g. `{"thermal": "lumped"}`) to turn on the temperature model. Otherwise temperature stays constant.

**Reply by YouHyin** [reply]
I have another questions about that.

![figure-1-2](https://user-images.githubusercontent.com/113442395/193288778-0402c1df-c7f2-4c83-bf3e-a3844f79dfa3.png)

1. solution["Volume-averaged cell temperature [K]"].Entries -> I want to know how the data is stored here.

2. I wonder if the temperature according to the time can be brought up in the form of an array, that is, I want to see the temperature change according to time as data.

3. I wonder if I can break and leave if certain conditions are satisfied within the experimental function. For example, when the cell temperature reaches 40 degrees, I want to break the experiment.

Thank you always!

**Reply by valentinsulzer** [reply]
1. It's a numpy array
2. `solution["Volume-averaged cell temperature [K]"](t)` gives you the temperature at time `t`. `t` can be a number or numpy vector
3. This is not possible

**Reply by YouHyin** [reply]
I solved it with your help in the past. But there's one question that's left now.
![temperature](https://user-images.githubusercontent.com/113442395/199183430-1a735bbe-e04e-42cb-bead-1c6eae37326e.png)
<Fig.1>

But I think theoretically the temperature should drop over time. (Figure 2) Looking at the result plot, the temperature does not change as it converges to the maximum value over time. What should I do to make the temperature drop over time as shown in Figure 2?

![tmep1](https://user-images.githubusercontent.com/113442395/199233117-ff4d0813-ebfb-49b2-8b11-74b11a001be5.png)
<Fig.2>
